(cross from x86_64-linux) systems.examples.raspberryPi: zsh fails to build

### Describe the bug

This is from nixos-unstable current. For some reason, I get build failures with ZSH.

This is with a nixpkgs (system=x86_64-linux) and nixpkgs.crossSystem set to `systems.examples.raspberryPi`.

This seems to be the same failure (I checked, its the same derivation):
```
nix build .\#legacyPackages.x86_64-linux.pkgsCross.raspberryPi.zsh
```


```
building '/nix/store/l95j31dy5qqlijsyjprhwyfxrl10j2ms-zsh-armv6l-unknown-linux-gnueabihf-5.8.1.drv'...
[trim]
zsh-armv6l-unknown-linux-gnueabihf> make[4]: Entering directory '/build/zsh-5.8.1/Src/Zle'
zsh-armv6l-unknown-linux-gnueabihf> make[4]: Leaving directory '/build/zsh-5.8.1/Src/Zle'
zsh-armv6l-unknown-linux-gnueabihf> Updated `zleparameter.mdh'.
zsh-armv6l-unknown-linux-gnueabihf> echo 'timestamp for zleparameter.mdh against zleparameter.mdd' > zleparameter.mdhs
zsh-armv6l-unknown-linux-gnueabihf> gawk -f ../../Src/makepro.awk zleparameter.c Src/Zle > zleparameter.syms
zsh-armv6l-unknown-linux-gnueabihf> (echo '/* Generated automatically */'; sed -n '/^E/{s/^E//;p;}' < zleparameter.syms) \
zsh-armv6l-unknown-linux-gnueabihf>     > zleparameter.epro
zsh-armv6l-unknown-linux-gnueabihf> (echo '/* Generated automatically */'; sed -n '/^L/{s/^L//;p;}' < zleparameter.syms) \
zsh-armv6l-unknown-linux-gnueabihf>     > `echo zleparameter.epro | sed 's/\.epro$/.pro/'`
zsh-armv6l-unknown-linux-gnueabihf> make[4]: Entering directory '/build/zsh-5.8.1/Src/Zle'
zsh-armv6l-unknown-linux-gnueabihf> make[4]: Leaving directory '/build/zsh-5.8.1/Src/Zle'
zsh-armv6l-unknown-linux-gnueabihf> Updated `zleparameter.mdh'.
zsh-armv6l-unknown-linux-gnueabihf> make[3]: Leaving directory '/build/zsh-5.8.1/Src/Zle'
zsh-armv6l-unknown-linux-gnueabihf> make[2]: Leaving directory '/build/zsh-5.8.1/Src'
zsh-armv6l-unknown-linux-gnueabihf> rm -f stamp-modobjs.tmp
zsh-armv6l-unknown-linux-gnueabihf> make[2]: Entering directory '/build/zsh-5.8.1/Src'
zsh-armv6l-unknown-linux-gnueabihf> armv6l-unknown-linux-gnueabihf-gcc -c -I. -I../Src -I../Src -I../Src/Zle -I. -I/nix/store/psizgq9cjgl18bkwmhxj8ma5ak791qwa-pcre-8.45-dev/include -DHAVE_CONFIG_H -Wall -Wmissing-prototypes -O2  -o builtin.o builtin.c
zsh-armv6l-unknown-linux-gnueabihf> In file included from zsh.mdh:15,
zsh-armv6l-unknown-linux-gnueabihf>                  from builtin.c:33:
zsh-armv6l-unknown-linux-gnueabihf> zsh_system.h:490:3: error: #error "Don't know how to change UID"
zsh-armv6l-unknown-linux-gnueabihf>   490 | # error "Don't know how to change UID"
zsh-armv6l-unknown-linux-gnueabihf>       |   ^~~~~
zsh-armv6l-unknown-linux-gnueabihf> zsh_system.h:494:3: error: #error "Don't know how to change GID"
zsh-armv6l-unknown-linux-gnueabihf>   494 | # error "Don't know how to change GID"
zsh-armv6l-unknown-linux-gnueabihf>       |   ^~~~~
zsh-armv6l-unknown-linux-gnueabihf> In file included from zsh.mdh:19,
zsh-armv6l-unknown-linux-gnueabihf>                  from builtin.c:33:
zsh-armv6l-unknown-linux-gnueabihf> prototypes.h:118:12: error: conflicting types for 'gettimeofday'
zsh-armv6l-unknown-linux-gnueabihf>   118 | extern int gettimeofday _((struct timeval *tv, struct timezone *tz));
zsh-armv6l-unknown-linux-gnueabihf>       |            ^~~~~~~~~~~~
zsh-armv6l-unknown-linux-gnueabihf> In file included from zsh_system.h:239,
zsh-armv6l-unknown-linux-gnueabihf>                  from zsh.mdh:15,
zsh-armv6l-unknown-linux-gnueabihf>                  from builtin.c:33:
zsh-armv6l-unknown-linux-gnueabihf> /nix/store/4f8rzcciwklkfwgsp1a683ing2nn159q-armv6l-unknown-linux-gnueabihf-stage-final-gcc-debug-10.3.0/armv6l-unknown-linux-gnueabihf/sys-include/sys/time.h:67:12: note: previous declaration of 'gettimeofday' was here
zsh-armv6l-unknown-linux-gnueabihf>    67 | extern int gettimeofday (struct timeval *__restrict __tv,
zsh-armv6l-unknown-linux-gnueabihf>       |            ^~~~~~~~~~~~
zsh-armv6l-unknown-linux-gnueabihf> In file included from zsh.mdh:33,
zsh-armv6l-unknown-linux-gnueabihf>                  from builtin.c:33:
zsh-armv6l-unknown-linux-gnueabihf> compat.epro:12:12: error: conflicting types for 'gettimeofday'
zsh-armv6l-unknown-linux-gnueabihf>    12 | extern int gettimeofday _((struct timeval*tv,struct timezone*tz));
zsh-armv6l-unknown-linux-gnueabihf>       |            ^~~~~~~~~~~~
zsh-armv6l-unknown-linux-gnueabihf> In file included from zsh_system.h:239,
zsh-armv6l-unknown-linux-gnueabihf>                  from zsh.mdh:15,
zsh-armv6l-unknown-linux-gnueabihf>                  from builtin.c:33:
zsh-armv6l-unknown-linux-gnueabihf> /nix/store/4f8rzcciwklkfwgsp1a683ing2nn159q-armv6l-unknown-linux-gnueabihf-stage-final-gcc-debug-10.3.0/armv6l-unknown-linux-gnueabihf/sys-include/sys/time.h:67:12: note: previous declaration of 'gettimeofday' was here
zsh-armv6l-unknown-linux-gnueabihf>    67 | extern int gettimeofday (struct timeval *__restrict __tv,
zsh-armv6l-unknown-linux-gnueabihf>       |            ^~~~~~~~~~~~
zsh-armv6l-unknown-linux-gnueabihf> In file included from zsh.mdh:33,
zsh-armv6l-unknown-linux-gnueabihf>                  from builtin.c:33:
zsh-armv6l-unknown-linux-gnueabihf> compat.epro:32:34: error: expected ')' before ';' token
zsh-armv6l-unknown-linux-gnueabihf>    32 | extern unsigned long strtoul(nptr;
zsh-armv6l-unknown-linux-gnueabihf>       |                                  ^
zsh-armv6l-unknown-linux-gnueabihf>       |                                  )
zsh-armv6l-unknown-linux-gnueabihf> In file included from zsh.mdh:15,
zsh-armv6l-unknown-linux-gnueabihf>                  from builtin.c:33:
zsh-armv6l-unknown-linux-gnueabihf> builtin.c: In function 'new_optarg':
zsh-armv6l-unknown-linux-gnueabihf> zsh_system.h:789:53: warning: assignment to 'char *' from incompatible pointer type 'char **' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types8;;]
zsh-armv6l-unknown-linux-gnueabihf>   789 | # define memmove(dest, src, len) (bcopy((src), zmmv = (dest), (len)), zmmv)
zsh-armv6l-unknown-linux-gnueabihf>       |                                                     ^
zsh-armv6l-unknown-linux-gnueabihf> zsh_system.h:783:17: note: in expansion of macro 'memmove'
zsh-armv6l-unknown-linux-gnueabihf>   783 | # define memcpy memmove
zsh-armv6l-unknown-linux-gnueabihf>       |                 ^~~~~~~
zsh-armv6l-unknown-linux-gnueabihf> builtin.c:238:6: note: in expansion of macro 'memcpy'
zsh-armv6l-unknown-linux-gnueabihf>   238 |      memcpy(newptr, ops->args, ops->argsalloc * sizeof(char *));
zsh-armv6l-unknown-linux-gnueabihf>       |      ^~~~~~
zsh-armv6l-unknown-linux-gnueabihf> zsh_system.h:789:69: warning: right-hand operand of comma expression has no effect [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-value-Wunused-value8;;]
zsh-armv6l-unknown-linux-gnueabihf>   789 | # define memmove(dest, src, len) (bcopy((src), zmmv = (dest), (len)), zmmv)
zsh-armv6l-unknown-linux-gnueabihf>       |                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
zsh-armv6l-unknown-linux-gnueabihf> zsh_system.h:783:17: note: in expansion of macro 'memmove'
zsh-armv6l-unknown-linux-gnueabihf>   783 | # define memcpy memmove
zsh-armv6l-unknown-linux-gnueabihf>       |                 ^~~~~~~
zsh-armv6l-unknown-linux-gnueabihf> builtin.c:238:6: note: in expansion of macro 'memcpy'
zsh-armv6l-unknown-linux-gnueabihf>   238 |      memcpy(newptr, ops->args, ops->argsalloc * sizeof(char *));
zsh-armv6l-unknown-linux-gnueabihf>       |      ^~~~~~
zsh-armv6l-unknown-linux-gnueabihf> builtin.c: In function 'bin_functions':
zsh-armv6l-unknown-linux-gnueabihf> zsh_system.h:789:53: warning: assignment to 'char *' from incompatible pointer type 'Shfunc' {aka 'struct shfunc *'} [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types8;;]
zsh-armv6l-unknown-linux-gnueabihf>   789 | # define memmove(dest, src, len) (bcopy((src), zmmv = (dest), (len)), zmmv)
zsh-armv6l-unknown-linux-gnueabihf>       |                                                     ^
zsh-armv6l-unknown-linux-gnueabihf> zsh_system.h:783:17: note: in expansion of macro 'memmove'
zsh-armv6l-unknown-linux-gnueabihf>   783 | # define memcpy memmove
zsh-armv6l-unknown-linux-gnueabihf>       |                 ^~~~~~~
zsh-armv6l-unknown-linux-gnueabihf> builtin.c:3279:2: note: in expansion of macro 'memcpy'
zsh-armv6l-unknown-linux-gnueabihf>  3279 |  memcpy(newsh, shf, sizeof(*newsh));
zsh-armv6l-unknown-linux-gnueabihf>       |  ^~~~~~
zsh-armv6l-unknown-linux-gnueabihf> zsh_system.h:789:69: warning: right-hand operand of comma expression has no effect [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-value-Wunused-value8;;]
zsh-armv6l-unknown-linux-gnueabihf>   789 | # define memmove(dest, src, len) (bcopy((src), zmmv = (dest), (len)), zmmv)
zsh-armv6l-unknown-linux-gnueabihf>       |                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
zsh-armv6l-unknown-linux-gnueabihf> zsh_system.h:783:17: note: in expansion of macro 'memmove'
zsh-armv6l-unknown-linux-gnueabihf>   783 | # define memcpy memmove
zsh-armv6l-unknown-linux-gnueabihf>       |                 ^~~~~~~
zsh-armv6l-unknown-linux-gnueabihf> builtin.c:3279:2: note: in expansion of macro 'memcpy'
zsh-armv6l-unknown-linux-gnueabihf>  3279 |  memcpy(newsh, shf, sizeof(*newsh));
zsh-armv6l-unknown-linux-gnueabihf>       |  ^~~~~~
zsh-armv6l-unknown-linux-gnueabihf> builtin.c: In function 'bin_print':
zsh-armv6l-unknown-linux-gnueabihf> zsh_system.h:789:53: warning: assignment to 'char *' from incompatible pointer type 'short int *' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types8;;]
zsh-armv6l-unknown-linux-gnueabihf>   789 | # define memmove(dest, src, len) (bcopy((src), zmmv = (dest), (len)), zmmv)
zsh-armv6l-unknown-linux-gnueabihf>       |                                                     ^
zsh-armv6l-unknown-linux-gnueabihf> zsh_system.h:783:17: note: in expansion of macro 'memmove'
zsh-armv6l-unknown-linux-gnueabihf>   783 | # define memcpy memmove
zsh-armv6l-unknown-linux-gnueabihf>       |                 ^~~~~~~
zsh-armv6l-unknown-linux-gnueabihf> builtin.c:4898:7: note: in expansion of macro 'memcpy'
zsh-armv6l-unknown-linux-gnueabihf>  4898 |       memcpy(ent->words, words, nwords*sizeof(short));
zsh-armv6l-unknown-linux-gnueabihf>       |       ^~~~~~
zsh-armv6l-unknown-linux-gnueabihf> zsh_system.h:789:69: warning: right-hand operand of comma expression has no effect [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-value-Wunused-value8;;]
zsh-armv6l-unknown-linux-gnueabihf>   789 | # define memmove(dest, src, len) (bcopy((src), zmmv = (dest), (len)), zmmv)
zsh-armv6l-unknown-linux-gnueabihf>       |                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
zsh-armv6l-unknown-linux-gnueabihf> zsh_system.h:783:17: note: in expansion of macro 'memmove'
zsh-armv6l-unknown-linux-gnueabihf>   783 | # define memcpy memmove
zsh-armv6l-unknown-linux-gnueabihf>       |                 ^~~~~~~
zsh-armv6l-unknown-linux-gnueabihf> builtin.c:4898:7: note: in expansion of macro 'memcpy'
zsh-armv6l-unknown-linux-gnueabihf>  4898 |       memcpy(ent->words, words, nwords*sizeof(short));
zsh-armv6l-unknown-linux-gnueabihf>       |       ^~~~~~
zsh-armv6l-unknown-linux-gnueabihf> builtin.c: In function 'bin_shift':
zsh-armv6l-unknown-linux-gnueabihf> zsh_system.h:789:53: warning: assignment to 'char *' from incompatible pointer type 'char **' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types8;;]
zsh-armv6l-unknown-linux-gnueabihf>   789 | # define memmove(dest, src, len) (bcopy((src), zmmv = (dest), (len)), zmmv)
zsh-armv6l-unknown-linux-gnueabihf>       |                                                     ^
zsh-armv6l-unknown-linux-gnueabihf> zsh_system.h:783:17: note: in expansion of macro 'memmove'
zsh-armv6l-unknown-linux-gnueabihf>   783 | # define memcpy memmove
zsh-armv6l-unknown-linux-gnueabihf>       |                 ^~~~~~~
zsh-armv6l-unknown-linux-gnueabihf> builtin.c:5472:3: note: in expansion of macro 'memcpy'
zsh-armv6l-unknown-linux-gnueabihf>  5472 |   memcpy(s, pparams, (l - num) * sizeof(char *));
zsh-armv6l-unknown-linux-gnueabihf>       |   ^~~~~~
zsh-armv6l-unknown-linux-gnueabihf> zsh_system.h:789:69: warning: right-hand operand of comma expression has no effect [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-value-Wunused-value8;;]
zsh-armv6l-unknown-linux-gnueabihf>   789 | # define memmove(dest, src, len) (bcopy((src), zmmv = (dest), (len)), zmmv)
zsh-armv6l-unknown-linux-gnueabihf>       |                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
zsh-armv6l-unknown-linux-gnueabihf> zsh_system.h:783:17: note: in expansion of macro 'memmove'
zsh-armv6l-unknown-linux-gnueabihf>   783 | # define memcpy memmove
zsh-armv6l-unknown-linux-gnueabihf>       |                 ^~~~~~~
zsh-armv6l-unknown-linux-gnueabihf> builtin.c:5472:3: note: in expansion of macro 'memcpy'
zsh-armv6l-unknown-linux-gnueabihf>  5472 |   memcpy(s, pparams, (l - num) * sizeof(char *));
zsh-armv6l-unknown-linux-gnueabihf>       |   ^~~~~~
zsh-armv6l-unknown-linux-gnueabihf> zsh_system.h:789:53: warning: assignment to 'char *' from incompatible pointer type 'char **' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types8;;]
zsh-armv6l-unknown-linux-gnueabihf>   789 | # define memmove(dest, src, len) (bcopy((src), zmmv = (dest), (len)), zmmv)
zsh-armv6l-unknown-linux-gnueabihf>       |                                                     ^
zsh-armv6l-unknown-linux-gnueabihf> zsh_system.h:783:17: note: in expansion of macro 'memmove'
zsh-armv6l-unknown-linux-gnueabihf>   783 | # define memcpy memmove
zsh-armv6l-unknown-linux-gnueabihf>       |                 ^~~~~~~
zsh-armv6l-unknown-linux-gnueabihf> builtin.c:5477:3: note: in expansion of macro 'memcpy'
zsh-armv6l-unknown-linux-gnueabihf>  5477 |   memcpy(s, pparams + num, (l - num + 1) * sizeof(char *));
zsh-armv6l-unknown-linux-gnueabihf>       |   ^~~~~~
zsh-armv6l-unknown-linux-gnueabihf> zsh_system.h:789:69: warning: right-hand operand of comma expression has no effect [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-value-Wunused-value8;;]
zsh-armv6l-unknown-linux-gnueabihf>   789 | # define memmove(dest, src, len) (bcopy((src), zmmv = (dest), (len)), zmmv)
zsh-armv6l-unknown-linux-gnueabihf>       |                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
zsh-armv6l-unknown-linux-gnueabihf> zsh_system.h:783:17: note: in expansion of macro 'memmove'
zsh-armv6l-unknown-linux-gnueabihf>   783 | # define memcpy memmove
zsh-armv6l-unknown-linux-gnueabihf>       |                 ^~~~~~~
zsh-armv6l-unknown-linux-gnueabihf> builtin.c:5477:3: note: in expansion of macro 'memcpy'
zsh-armv6l-unknown-linux-gnueabihf>  5477 |   memcpy(s, pparams + num, (l - num + 1) * sizeof(char *));
zsh-armv6l-unknown-linux-gnueabihf>       |   ^~~~~~
zsh-armv6l-unknown-linux-gnueabihf> builtin.c: In function 'bin_emulate':
zsh-armv6l-unknown-linux-gnueabihf> zsh_system.h:789:69: warning: right-hand operand of comma expression has no effect [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-value-Wunused-value8;;]
zsh-armv6l-unknown-linux-gnueabihf>   789 | # define memmove(dest, src, len) (bcopy((src), zmmv = (dest), (len)), zmmv)
zsh-armv6l-unknown-linux-gnueabihf>       |                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
zsh-armv6l-unknown-linux-gnueabihf> zsh_system.h:783:17: note: in expansion of macro 'memmove'
zsh-armv6l-unknown-linux-gnueabihf>   783 | # define memcpy memmove
zsh-armv6l-unknown-linux-gnueabihf>       |                 ^~~~~~~
zsh-armv6l-unknown-linux-gnueabihf> builtin.c:6096:6: note: in expansion of macro 'memcpy'
zsh-armv6l-unknown-linux-gnueabihf>  6096 |      memcpy(cmdopts, opts, OPT_SIZE);
zsh-armv6l-unknown-linux-gnueabihf>       |      ^~~~~~
zsh-armv6l-unknown-linux-gnueabihf> zsh_system.h:789:69: warning: right-hand operand of comma expression has no effect [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-value-Wunused-value8;;]
zsh-armv6l-unknown-linux-gnueabihf>   789 | # define memmove(dest, src, len) (bcopy((src), zmmv = (dest), (len)), zmmv)
zsh-armv6l-unknown-linux-gnueabihf>       |                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
zsh-armv6l-unknown-linux-gnueabihf> zsh_system.h:783:17: note: in expansion of macro 'memmove'
zsh-armv6l-unknown-linux-gnueabihf>   783 | # define memcpy memmove
zsh-armv6l-unknown-linux-gnueabihf>       |                 ^~~~~~~
zsh-armv6l-unknown-linux-gnueabihf> builtin.c:6117:5: note: in expansion of macro 'memcpy'
zsh-armv6l-unknown-linux-gnueabihf>  6117 |     memcpy(saveopts, opts, sizeof(opts));
zsh-armv6l-unknown-linux-gnueabihf>       |     ^~~~~~
zsh-armv6l-unknown-linux-gnueabihf> zsh_system.h:789:69: warning: right-hand operand of comma expression has no effect [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-value-Wunused-value8;;]
zsh-armv6l-unknown-linux-gnueabihf>   789 | # define memmove(dest, src, len) (bcopy((src), zmmv = (dest), (len)), zmmv)
zsh-armv6l-unknown-linux-gnueabihf>       |                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
zsh-armv6l-unknown-linux-gnueabihf> zsh_system.h:783:17: note: in expansion of macro 'memmove'
zsh-armv6l-unknown-linux-gnueabihf>   783 | # define memcpy memmove
zsh-armv6l-unknown-linux-gnueabihf>       |                 ^~~~~~~
zsh-armv6l-unknown-linux-gnueabihf> builtin.c:6118:5: note: in expansion of macro 'memcpy'
zsh-armv6l-unknown-linux-gnueabihf>  6118 |     memcpy(new_opts, opts, sizeof(opts));
zsh-armv6l-unknown-linux-gnueabihf>       |     ^~~~~~
zsh-armv6l-unknown-linux-gnueabihf> zsh_system.h:789:69: warning: right-hand operand of comma expression has no effect [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-value-Wunused-value8;;]
zsh-armv6l-unknown-linux-gnueabihf>   789 | # define memmove(dest, src, len) (bcopy((src), zmmv = (dest), (len)), zmmv)
zsh-armv6l-unknown-linux-gnueabihf>       |                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
zsh-armv6l-unknown-linux-gnueabihf> zsh_system.h:783:17: note: in expansion of macro 'memmove'
zsh-armv6l-unknown-linux-gnueabihf>   783 | # define memcpy memmove
zsh-armv6l-unknown-linux-gnueabihf>       |                 ^~~~~~~
zsh-armv6l-unknown-linux-gnueabihf> builtin.c:6142:5: note: in expansion of macro 'memcpy'
zsh-armv6l-unknown-linux-gnueabihf>  6142 |     memcpy(opts, new_opts, sizeof(opts));
zsh-armv6l-unknown-linux-gnueabihf>       |     ^~~~~~
zsh-armv6l-unknown-linux-gnueabihf> zsh_system.h:789:69: warning: right-hand operand of comma expression has no effect [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-value-Wunused-value8;;]
zsh-armv6l-unknown-linux-gnueabihf>   789 | # define memmove(dest, src, len) (bcopy((src), zmmv = (dest), (len)), zmmv)
zsh-armv6l-unknown-linux-gnueabihf>       |                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
zsh-armv6l-unknown-linux-gnueabihf> zsh_system.h:783:17: note: in expansion of macro 'memmove'
zsh-armv6l-unknown-linux-gnueabihf>   783 | # define memcpy memmove
zsh-armv6l-unknown-linux-gnueabihf>       |                 ^~~~~~~
zsh-armv6l-unknown-linux-gnueabihf> builtin.c:6192:5: note: in expansion of macro 'memcpy'
zsh-armv6l-unknown-linux-gnueabihf>  6192 |     memcpy(opts, saveopts, sizeof(opts));
zsh-armv6l-unknown-linux-gnueabihf>       |     ^~~~~~
zsh-armv6l-unknown-linux-gnueabihf> make[2]: *** [Makemod:230: builtin.o] Error 1
zsh-armv6l-unknown-linux-gnueabihf> make[2]: Leaving directory '/build/zsh-5.8.1/Src'
zsh-armv6l-unknown-linux-gnueabihf> make[1]: *** [Makefile:449: modobjs] Error 2
zsh-armv6l-unknown-linux-gnueabihf> make[1]: Leaving directory '/build/zsh-5.8.1/Src'
zsh-armv6l-unknown-linux-gnueabihf> make: *** [Makefile:188: all] Error 1
error: builder for '/nix/store/l95j31dy5qqlijsyjprhwyfxrl10j2ms-zsh-armv6l-unknown-linux-gnueabihf-5.8.1.drv' failed with exit code 2;
       last 10 log lines:
       >   783 | # define memcpy memmove
       >       |                 ^~~~~~~
       > builtin.c:6192:5: note: in expansion of macro 'memcpy'
       >  6192 |     memcpy(opts, saveopts, sizeof(opts));
       >       |     ^~~~~~
       > make[2]: *** [Makemod:230: builtin.o] Error 1
       > make[2]: Leaving directory '/build/zsh-5.8.1/Src'
       > make[1]: *** [Makefile:449: modobjs] Error 2
       > make[1]: Leaving directory '/build/zsh-5.8.1/Src'
       > make: *** [Makefile:188: all] Error 1
       For full logs, run 'nix log /nix/store/l95j31dy5qqlijsyjprhwyfxrl10j2ms-zsh-armv6l-unknown-linux-gnueabihf-5.8.1.drv'.

```

### Steps To Reproduce
Steps to reproduce the behavior:
1. ...
2. ...
3. ...

### Expected behavior
A clear and concise description of what you expected to happen.

### Screenshots
If applicable, add screenshots to help explain your problem.

### Additional context
Add any other context about the problem here.

### Notify maintainers
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here
```

--- a/pkgs/shells/zsh/default.nix
+++ b/pkgs/shells/zsh/default.nix
@@ -10,6 +10,7 @@
 , texinfo
 , ncurses
 , pcre
+, pkg-config
 , buildPackages }:
 
 let
@@ -32,7 +33,7 @@ stdenv.mkDerivation {
   ];
 
   strictDeps = true;
-  nativeBuildInputs = [ autoreconfHook perl groff texinfo pcre]
+  nativeBuildInputs = [ autoreconfHook perl groff texinfo pkg-config ]
                       ++ lib.optionals stdenv.isLinux [ util-linux yodl ];
 
   buildInputs = [ ncurses pcre ];
@@ -55,6 +56,11 @@ stdenv.mkDerivation {
     "zsh_cv_sys_dynamic_strip_exe=yes"
     "zsh_cv_sys_dynamic_strip_lib=yes"
   ];
+
+  preConfigure = ''
+    # use pkg-config instead of pcre-config
+    configureFlagsArray+=("PCRECONF=''${PKG_CONFIG} libpcre")
+  '';
 
   # the zsh/zpty module is not available on hydra
   # so skip groups Y Z

